src/routeCombinators.ts

```diff
@@ -20,6 +30,93 @@ export const lit = (literal: string): Route => ({
   parts: [routeLiteral(literal)],
 })
 
+export function routes(parts: TemplateStringsArray): Route
```

Comment:
This hellscape probably deserves it's own file.

Reply:
Is using rest args not possible?

Reply:
It's fine for the implementation, but at the type-level it's hard to describe "here is a list of types, inductively combine them all with the `CR` type-level function to give me the return type" without having explicit types to talk about.